app/controllers/devise/two_factor_authentication_controller.rb

```diff
@@ -27,6 +27,11 @@ def update
     end
   end
 
+  def resend_code
+    resource.send_two_factor_authentication_code
+    redirect_to user_two_factor_authentication_path, notice: "Your authentication code has been sent."
```

Comment:
What do you think about localizing this notice, so that people can customize it?